cudamapper/src/index_generator_gpu.cu

```diff
@@ -753,309 +757,413 @@ namespace claragenomics {
-        read_id_to_compressed_minimizers_d.free();
-        CGA_LOG_INFO("Deallocating {} bytes from read_id_to_windows_section_d", read_id_to_windows_section_d.size()*sizeof(decltype(read_id_to_windows_section_d)::value_type));
-        read_id_to_windows_section_d.free();
-    
-        // *** sort minimizers by representation ***
-        // As this is a stable sort and the data was initailly grouper by read_id this means that the minimizers within each representations are sorted by read_id
-        thrust::stable_sort_by_key(thrust::device, representations_compressed_d.data(), representations_compressed_d.data() + total_minimizers, rest_compressed_d.data());
-
-        std::vector<representation_t> representations_compressed_h(total_minimizers);
-        std::vector<ReadidPositionDirection> rest_compressed_h(total_minimizers);
-        CGA_CU_CHECK_ERR(cudaMemcpy(representations_compressed_h.data(),
-                                    representations_compressed_d.data(),
-                                    representations_compressed_h.size()*sizeof(decltype(representations_compressed_h)::value_type),
-                                    cudaMemcpyDeviceToHost
-                                    )
-                         );
-        CGA_CU_CHECK_ERR(cudaMemcpy(rest_compressed_h.data(),
-                                    rest_compressed_d.data(),
-                                    rest_compressed_h.size()*sizeof(decltype(rest_compressed_h)::value_type),
-                                    cudaMemcpyDeviceToHost
-                                    )
-                         );
-
-        // free these arrays as they are not needed anymore
-        CGA_LOG_INFO("Deallocating {} bytes from representations_compressed_d", representations_compressed_d.size()*sizeof(decltype(representations_compressed_d)::value_type));
-        representations_compressed_d.free();
-        CGA_LOG_INFO("Deallocating {} bytes from rest_compressed_d", rest_compressed_d.size()*sizeof(decltype(rest_compressed_d)::value_type));
-        rest_compressed_d.free();
+            all_representation_readid_position_direction.push_back(representation_read_id_position_direction);
 
+            if (parser_status == false) {
+                break;
+            }
+        }
         // *** add the minimizers to the host side index ***
         // minimizers are already sorted by representation -> add all minimizers with the same representation to a vector and then add that vector to the hash table
+        std::vector<std::pair<representation_t, ReadidPositionDirection>> repr_rest_pairs;
+
+        //Lambda function to compare std::pair<representation_t, ReadidPositionDirection>
+        std::function<bool(std::pair<representation_t, ReadidPositionDirection>, std::pair<representation_t, ReadidPositionDirection>)> compare =
```

Comment:
If you change from `template <typename T> merge_n_sorted_vectors` to `template <typename T, Compare> merge_n_sorted_vectors` (see my comment in `cudamapper_utils.hpp`) you won't have to create `std::function` around lambda

Reply:
done